src/libraries/System.Private.Runtime.InteropServices.JavaScript/src/System/Runtime/InteropServices/JavaScript/Marshallers/JSObjectMarshaller.cs

```diff
@@ -0,0 +1,45 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Runtime.CompilerServices;
+
+namespace System.Runtime.InteropServices.JavaScript.Private
+{
+    /// <summary>
+    /// The JSObject instances roundtip and keep identity.
+    /// </summary>
+    internal class JSObjectMarshaller : BaseMarshaller<IJSObject>
+    {
+#nullable disable
+        public override MarshalToManagedDelegate ToManaged => (ref JavaScriptMarshallerArg arg) => JavaScriptMarshalImpl.MarshalToManaged(ref arg, default(IJSObject));
+        public override MarshalToJavaScriptDelegate ToJavaScript => (object value, ref JavaScriptMarshallerArg arg) => JavaScriptMarshalImpl.MarshalToJs((IJSObject)value, ref arg);
```

Comment:
overloading these methods by type and having to do the cast is kind of gross and magical, if the statics are essential to the marshaler working let's give them distinct names i.e. MarshalToJs_JSObject or MarshalToJs_Exception

Reply:
To be clearer I think the overload is fine for primitives, but there's nothing stopping an Exception from also being an IJSObject type system wise so it's unwise to rely on overloads here

Reply:
Agree with reverting the overloads. 
That gross `object` cast is gross, but it could not be prevented, in order to keep common signature.
Luckily, this would not be an issue for the roslyn generated code and this dynamic resolution would not be there at all.